### DISCUSSION TITLE
suggestion : add user permission delete_dirs

### DISCUSSION THREAD

**Discussion by guedz** [drakkan/sftpgo#600] [Ideas]
Hello,

I unfortunately have a few users who delete the destination folder for their upload, which then generates errors on their side and therefore on the support.
Is it possible to add the possibility to prohibit the deletion of folder? 

Best regards

**Comment by drakkan** [comment]
Hi,

supposing your upload directory is `/upload` you can remove the delete permission on `/` and allow any permission for `/upload`. This should solve your issue if I understand it correctly

**Comment by guedz45** [comment]
Hi,

in general i have many cases.
In example, /upload/invoices, /upload/pdf, /upload/invoices/xml etc... 
Manage with linux, windows, macos folder permissions can be difficult.
If we have the permission create_dir, it's for me more simple to have delete_dir (as https://mysecureshell.readthedocs.io/en/latest/tags/childs/disableremovedir.html)
i think it"s more simple to disallow rmdir command than manage folder permissions and i need to have the possibility to delete file after download but not delete folder.
Best regards

**Reply by drakkan** [reply]
> Hi,
> 
> in general i have many cases. In example, /upload/invoices, /upload/pdf, /upload/invoices/xml etc... Manage with linux, windows, macos folder permissions can be difficult. If we have the permission create_dir, it's for me more simple to have delete_dir (as https://mysecureshell.readthedocs.io/en/latest/tags/childs/disableremovedir.html) i think it"s more simple to disallow rmdir command than manage folder permissions and i need to have the possibility to delete file after download but not delete folder. Best regards

I mean virtual permissions, please take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/howto/getting-started.md#Virtual-permissions). Permissions apply recursively so if you remove the `delete` permission you will not be able to delete anything if you don't override the permissions on a per-directorry basis. 

Currently we have a generic `delete` permissions that apply to both files and directories I could also evaluate to add more specific permissions, for example `delete_dir`, `delete_file` a sponsorship will help. Thanks

**Comment by drakkan** [comment]
Hi,

this is now supported https://github.com/drakkan/sftpgo/commit/ca730e77a5a1cbe95a641c60b6f07f3c2d0430af

**Comment by guedz45** [comment]
Hi Drakkan,
really thank you very much for your work. 
I will test all of this as soon as possible.
I take advantage of my post to ask if there are any limitations on the number of simultaneous connections, max accounts created with sftpgo.
In some solutions even with the parameter to 0 for unlimited there are limitations related to the code (example) as for SSH where the AllowUsers lines are limited in the code to 256 unless we modify the source code and we recompile ourselves . 

Thansk again
Best regards

**Reply by drakkan** [reply]
Hi,

the SFTPGo code has no limitations, you are limited from the hw resources.

I am not aware of such limitations in the libraries used in SFTPGo.

If you plan to have many concurrent client connections, on Linux you have to increase the open file limit. If you use the provided systemd unit this limit is [8192](https://github.com/drakkan/sftpgo/blob/main/init/sftpgo.service#L15) that should be enough for most use cases.
